src/caustics/tests.py

```diff
@@ -10,14 +10,24 @@
 
 __all__ = ["test"]
 
+# Pseudo device as mentioned in https://github.com/pytorch/pytorch/issues/61654
+# using this device, no actual computation is done, but will check
+# for where the tensors are located
+META_DEVICE = torch.device("meta")
 
-def _test_simulator_runs():
+TORCH_DEVICE = (
+    torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
+)
+
+
+def _test_simulator_runs(device=TORCH_DEVICE):
     # Model
     cosmology = FlatLambdaCDM(name="cosmo")
+    z_l = torch.tensor(1.0, device=device)
```

Comment:
Setting `z_l` to a device shouldn't be necessary because you have `sim.to` later. Though it seems that isn't working, right?

Reply:
I can revert these changes to see if that works out.